icaruscode/CRT/CRTTPCTruthEff_module.cc

```diff
@@ -0,0 +1,651 @@
+					temp_crt_trueid = bt.TrueIdFromTotalEnergy(event, crtcandidate);
+				 	GetAncestorID(temp_crt_trueid, temp_crt_motherid, temp_crt_ancestorid, temp_crt_motherlayers, particles);
+					all_crt_pdg.push_back(particles[temp_crt_trueid].PdgCode()); 
+	
+					all_crt_candidate_trueIDs.push_back(temp_crt_trueid);
+					all_crt_candidate_motherIDs.push_back(temp_crt_motherid);
+					all_crt_candidate_ancestorIDs.push_back(temp_crt_ancestorid);
+					all_crt_candidate_motherlayers.push_back(temp_crt_motherlayers);
+
+				}//end if(!fIsData)
+
+				all_crt_candidate_x.push_back(thiscand.crt_hit_pos.X());
+				all_crt_candidate_y.push_back(thiscand.crt_hit_pos.Y());
+				all_crt_candidate_z.push_back(thiscand.crt_hit_pos.Z());
+
+				all_crt_regions.push_back((int)fCrtutils->AuxDetRegionNameToNum(thiscand.thishit.tagger));
+				crt_timestamp.push_back(thiscand.crtTime);
+
+				crt_start_dca.push_back(thiscand.simpleDCA_startDir);	
+				crt_end_dca.push_back(thiscand.simpleDCA_endDir);	
+
+				all_crt_hitcode_map.push_back(i);//Gives ID to each hit for the FEB+channel+pes breakdown below to allow cross-referencing between vectors
+
+			    	std::map< uint8_t, std::vector<std::pair<int,float> > > thishit_pes = crtcandidate.pesmap; ///< Saves signal hit information (FEB, local-channel and PE) .
+				for(uint8_t k=1; k<232; k++){
+					if(thishit_pes[k].size()>0){
+						for(int j=0; j<(int)thishit_pes[k].size(); j++){
+							all_crt_mac5s.push_back(k);
+							all_crt_hitcode.push_back(i);
+							all_crt_chans.push_back(thishit_pes[k][j].first);
+							all_crt_pes.push_back(thishit_pes[k][j].second);
+						}//end for(int j=0; j<crt_pes[k].size(); j++)
+					}//end if(thishit_pes[k].size()>0)
+				}//end loop searching for FEBs connected to the hit
+			}//end loop over CRT candidates
+		}//end if(num_crt_candidates>0)
+
+	    }//end if(closest.dca != -99999)
+
+	  } // DCA check
+	  
+   	    tr_crttpc->Fill();
+	} // Loop over tracks  
+	
+      } // Validity check
+
+    } // all track labels in a vector 
+  } // CRTTPCTruthEff::produce()
+
+
+  void CRTTPCTruthEff::endJob()
+  {
+
+  } // CRTTPCTruthEff::endJob()
+
+void icarus::CRTTPCTruthEff::GetAncestorID(int trueid, int &motherid, int &ancestorid, int &layers, std::map<int, simb::MCParticle> all_particles){
+
+	int size_part = (int)all_particles.size();
+	if(trueid<size_part) ancestorid = all_particles[trueid].Mother();
+	else ancestorid = trueid;
+	motherid = ancestorid;
+	int last_ancestorid;
+	bool hitbottom = false;
+	int counter = 0;
+	while(ancestorid%10000000!=0 && counter<1000 && !hitbottom){
+		last_ancestorid = ancestorid;
+		if(ancestorid<size_part) ancestorid = all_particles[ancestorid].Mother();
+		hitbottom = last_ancestorid == ancestorid;
+		counter++;
+	}//end while loop 
+	layers = counter;
+
+
+}//CRTTPCTruthEff::GetAncestorID
+
+void icarus::CRTTPCTruthEff::getCatCrossXYZ(recob::Track trk, double &my_x, double &my_y, double &my_z){
```

Comment:
This may be deprecated later, getCatCrossXYZ was just something I made to see what my own guess would compare to the one provided by the track but I don't really need it these days